Classes/Utility/FLEXRuntimeUtility.m

```diff
@@ -277,14 +277,16 @@ + (NSArray *)prettyArgumentComponentsForMethod:(Method)method
     NSMutableArray *components = [NSMutableArray array];
     
     NSString *selectorName = NSStringFromSelector(method_getName(method));
-    NSArray *selectorComponents = [selectorName componentsSeparatedByString:@":"];
-    unsigned int numberOfArguments = method_getNumberOfArguments(method);
+    NSMutableArray *selectorComponents = [[selectorName componentsSeparatedByString:@":"] mutableCopy];
+    if ([selectorComponents.lastObject isEqualToString:@""]) {
+        [selectorComponents removeLastObject];
+    }
     
-    for (unsigned int argIndex = kFLEXNumberOfImplicitArgs; argIndex < numberOfArguments; argIndex++) {
+    for (unsigned int argIndex = 0; argIndex < selectorComponents.count; argIndex++) {
         char *argType = method_copyArgumentType(method, argIndex);
```

Comment:
Since the `method_getNumberOfArguments()` returns not the correct number for some methods, namely `URLSession:downloadTask:didFinishDownloadingToURL`, I suggest that instead we get selector components and run the loop for this array. In this case there the crash connected with out of bounds of array will be eliminated. Also, as in the `for` loop you get the object at index `argIndex - kFLEXNumberOfImplicitArgs` for selector components, it is the same as to start from 0 until the count of selector components, as done in my version.